llvm/lib/Transforms/InstCombine/InstCombineAndOrXor.cpp

```diff
@@ -4410,6 +4410,11 @@ Instruction *InstCombinerImpl::foldNot(BinaryOperator &I) {
   if (Instruction *NewXor = foldNotXor(I, Builder))
     return NewXor;
 
+  // TODO: Could handle multi-use better by checking if all uses of NotOp (other
+  // than I) can be inverted.
+  if (Value *R = getFreeInverted(NotOp, NotOp->hasOneUse(), &Builder))
+    return replaceInstUsesWith(I, R);
```

Comment:
There is a `sinkNotIntoXor()` fold that I think gets obsoleted by this one? Any test changes if you drop it?

Reply:
Some tests still rely on it, but did't poke around that deeply.

Reply:
Was actually just operand order changes. Dropping `sinkNotIntoXor`.